packages/loader/container-loader/src/containerContext.ts

```diff
@@ -331,7 +332,7 @@ export class ContainerContext implements IContainerContext {
             // If "module" is not in the result, we are using a legacy ICodeLoader.  Fix the result up with details.
             // Once usage drops to 0 we can remove this compat path.
             this.taggedLogger.sendTelemetryEvent({ eventName: "LegacyCodeLoader" });
-            return { module: loadCodeResult, details: codeDetails };
+            return { module: loadCodeResult as unknown as IFluidModule, details: codeDetails };
```

Comment:
Another way to do this is to say `const loadCodeResult = ... as IFluidModuleWithDetails | IFluidModule;` above, which correctly models reality:  `this.codeLoader.load` may return either format.  Then TS type narrowing correctly computes that `loadCodeResult` is of type `IFluidModule` here without any further casting.

I'd probably add a comment by the `as` above explaining about back compat if you go this way.

Reply:
That helps explain this. By adding `IFluidModule` as an option for `loadCodeResult` the else branch compiles now, no `as` needed anywhere. Typescript type narrowing is strange... I would have expected it to have inferred never on the else branch before and given a dead code error, instead no narrowing.